### PR TITLE
Close SSH session at the end of Provisioner.Run()

### PR DESCRIPTION
**Pull request by ArangoGutierrez**

### CODE REVIEW

**Review by elezar — APPROVED** [review]

File: pkg/provisioner/provisioner.go [hunk 1]
```diff
@@ -110,6 +110,11 @@ func (p *Provisioner) Run(env v1alpha1.Environment) error {
 		p.tpl.Reset()
 	}
 
+	// Close the current ssh connection
+	if err := p.Client.Close(); err != nil {
```
Comment (elezar) [inline pkg/provisioner/provisioner.go:114]: Question: Where do we actually open the connection? Should we not call close in a `defer` statement there?

Comment (ArangoGutierrez) [reply pkg/provisioner/provisioner.go:114]: we have a  function  `connectOrDie` to create the ssh session, and that func doesn't have the close
https://github.com/NVIDIA/holodeck/blob/main/pkg/provisioner/provisioner.go#L230

Comment (elezar) [reply pkg/provisioner/provisioner.go:114]:
```	if err = p.Run(opts.cfg); err != nil {
		return fmt.Errorf("failed to run provisioner: %v", err)
	}
	defer p.Client.Close()
```?

Comment (ArangoGutierrez) [reply pkg/provisioner/provisioner.go:114]: you mean setting the close at `cmd/cli/create/create.go` ? that's also valid

**Review by tariq1890 — APPROVED** [review]

**Pull request merged**